go/chat/bots/commands.go

```diff
@@ -343,20 +344,25 @@ func (b *CachingBotCommandManager) ListCommands(ctx context.Context, convID chat
 		return res, alias, nil
 	}
 
-	cmdDedup := make(map[string]bool)
+	cmdOutputs := make(map[string]chat1.UserBotCommandOutput)
+	cmdDedup := make(map[string]chat1.BotCommandsAdvertisementTyp)
 	for _, ad := range s.Advertisements {
 		ad.Username = libkb.NewNormalizedUsername(ad.Username).String()
 		if ad.Advertisement.Alias != nil {
 			alias[ad.Username] = *ad.Advertisement.Alias
 		}
 		for _, cmd := range ad.Advertisement.Commands {
 			key := cmd.Name + ad.Username
-			if !cmdDedup[key] {
-				res = append(res, cmd.ToOutput(ad.Username))
-				cmdDedup[key] = true
+			if typ, ok := cmdDedup[key]; !ok || chat1.AdvertRanking[ad.Typ] > chat1.AdvertRanking[typ] {
```

Comment:
can't we just compare the types directly? they are just an enum

Reply:
Yeah, my only concern is if we add new advertisement types they won't necessarily have their 'rank' the same as the enum value. We could worry about that later though, lmk what you think

Reply:
we can reintroduce this if we add a new type, can you drop a comment in the avdl? old clients won't handle the value missing from the map anyway